lnd.go

```diff
@@ -417,10 +419,61 @@ func getTLSConfig(cfg *config) (*tls.Config, *credentials.TransportCredentials,
 		return nil, nil, "", err
 	}
 
-	tlsConf = &tls.Config{
-		Certificates: []tls.Certificate{cert},
-		CipherSuites: tlsCipherSuites,
-		MinVersion:   tls.VersionTLS12,
+	// If Let's Encrypt is enabled, instantiate autocert to request/renew
+	// the certificates.
+	if cfg.LetsEncryptDomain != "" {
+		ltndLog.Infof(
+			"Using Let's Encrypt certificate for domain %v",
+			cfg.LetsEncryptDomain,
+		)
+		manager := autocert.Manager{
+			Cache:      autocert.DirCache(cfg.LetsEncryptDir),
+			Prompt:     autocert.AcceptTOS,
+			HostPolicy: autocert.HostWhitelist(cfg.LetsEncryptDomain),
+		}
+		go func() {
+			addr := fmt.Sprintf(":%v", cfg.LetsEncryptPort)
+
+			ltndLog.Infof(
+				"Autocert challenge listener started at %v",
+				addr,
+			)
+			err := http.ListenAndServe(
+				addr,
+				manager.HTTPHandler(nil),
+			)
+			if err != nil {
+				ltndLog.Errorf("autocert http: %v", err)
+			}
+		}()
+
+		getCertificate := func(h *tls.ClientHelloInfo) (
```

Comment:
can function be declared directly in the returned config? If not too messy.

Reply:
tried, but didn't like it too much